### DISCUSSION TITLE
3D UI

### DISCUSSION THREAD

**Discussion by SpaceJellyfishDev** [godotengine/godot-proposals#7729] [3D]
### Describe the project you are working on

I'm trying to create UI like this: Buttons have a sense of depth, individually move along Z axis when opening menu.
![IMG_20230917_220154](https://github.com/godotengine/godot-proposals/assets/144945239/d97743d8-edae-46a7-8fea-08221734ac49)

![IMG_20230917_215414](https://github.com/godotengine/godot-proposals/assets/144945239/068e2844-ff8f-4352-ab93-af235ac133b8)

### Describe the problem or limitation you are having in your project

I understand there is a demo project for building UI in 3D, but I find it very complicated if not impossible to create this effect. 

The current way for 3D UI is using camera texture on a mesh for every single moving UI. 
This would require 10-20 camera-mesh setup for a menu screen. And some cool UI features for organizing or aligning can't be used, since the meshes are in 3D space. 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

It would be great if we can place UI directly in 3D space.

It would allow more interesting UI and allow us to create cool 3D animations, such as having buttons/panels rotating and folding into places. 

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

I can't

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No

### Is there a reason why this should be core and not an add-on in the asset library?

UI is core

**Comment by Zireael07** [comment]
You misunderstand. You only need one camera. You need a mesh plus texture for every button. Also in 4.0 there's Label3D node which abstracts away the mesh+texture part.

**Comment by SpaceJellyfishDev** [comment]
I understand that one camera is enough, but it doesn't change the fact that 3D UI in godot is very complicated. 

As explained before:

containers, such as grid container, or other aligning/organizing features does not work in 3D space. 

resizing buttons/panels would also require resizing meshes. 

To achieve this 3D UI effect would be as complicated as rewriting UI. 

So, I would really want some features for creating 3D UI
